src/microbe_stage/MicrobeSpecies.cs

```diff
@@ -183,4 +187,21 @@ public override int GetVisualHashCode()
 
         return hash;
     }
+
+    public void ApplySceneParameters(Spatial instancedScene)
+    {
+        var microbe = (Microbe)instancedScene;
+        microbe.IsForPreviewOnly = true;
+
+        // We need to call _Ready here as the object may not be attached to the scene yet by the photo studio
+        microbe._Ready();
+
+        microbe.ApplySpecies(this);
+    }
+
+    public float CalculatePhotographDistance(Spatial instancedScene)
+    {
+        return PhotoStudio.CameraDistanceFromRadiusOfObject(((Microbe)instancedScene).Radius *
+            Constants.PHOTO_STUDIO_CELL_RADIUS_MULTIPLIER);
```

Comment:
This is duplicated code from MicrobeSpecies. MicrobeSpecies should be refactored to call these methods like this:
```
    public void ApplySceneParameters(Spatial instancedScene)
    {
       ((Microbe)instancedScene).ApplySceneParameters(instancedScene);
    }
```

Reply:
No wait sorry this is almost exactly duplicated code from CellType. That's the place these should be unified with as much as possible, especially bad is the long technical comment that is in multiple places.

Reply:
Wait, how do I apply species without telling Microbe what its species is?

Reply:
Because of that, I can't make Microbe photograph itself...

Reply:
Looking at the code, it looks like CellType creates a temporary microbe species object, so moving as much logic as possible to the MicrobeSpecies photograph would be optimal, then the CellType could just call those methods without having code duplication.